python/selfie-lib/selfie_lib/ArrayMap.py

```diff
@@ -15,107 +38,108 @@ def __len__(self) -> int: ...
     def __getitem__(self, index: Union[int, slice]) -> Union[T, List[T]]: ...
 
     def __contains__(self, item: object) -> bool:
-        for i in range(len(self)):
-            if self[i] == item:
-                return True
-        return False
+        try:
+            index = self.__binary_search(item)
+        except ValueError:
+            return False
```

Comment:
Why would there ever be a ValueError? `binarySearch` either returns negative values for missing values or it doesn't.